inferred-spans/src/test/java/co/elastic/otel/InferredSpansBackwardsCompatibilityConfigTest.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package co.elastic.otel;
+
+import static io.opentelemetry.sdk.testing.assertj.OpenTelemetryAssertions.assertThat;
+
+import co.elastic.otel.testing.AutoConfigTestProperties;
+import co.elastic.otel.testing.DisabledOnOpenJ9;
+import co.elastic.otel.testing.OtelReflectionUtils;
+import io.opentelemetry.api.GlobalOpenTelemetry;
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.contrib.inferredspans.InferredSpansProcessor;
+import io.opentelemetry.contrib.inferredspans.WildcardMatcher;
+import io.opentelemetry.contrib.inferredspans.internal.InferredSpansConfiguration;
+import io.opentelemetry.contrib.inferredspans.internal.ProfilingActivationListener;
+import io.opentelemetry.contrib.inferredspans.internal.SamplingProfiler;
+import io.opentelemetry.sdk.trace.SpanProcessor;
+import java.lang.reflect.Field;
+import java.nio.file.Path;
+import java.time.Duration;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+public class InferredSpansBackwardsCompatibilityConfigTest {
+
+  @BeforeEach
+  @AfterEach
+  public void resetGlobalOtel() {
+    ProfilingActivationListener.ensureInitialized();
+    OtelReflectionUtils.shutdownAndResetGlobalOtel();
+  }
+
+  @Test
+  @DisabledOnOpenJ9
+  public void checkAllLegacyOptions(@TempDir Path tmpDir) {
+    String libDir = tmpDir.resolve("foo").resolve("bar").toString();
+    try (AutoConfigTestProperties props =
+        new AutoConfigTestProperties()
+            .put("elastic.otel.inferred.spans.enabled", "true")
+            .put("elastic.otel.inferred.spans.logging.enabled", "false")
+            .put("elastic.otel.inferred.spans.backup.diagnostic.files", "true")
```

Comment:
maybe it would be worth adding a test to test the priority between legacy and upstream config options, focusing only on a few basic ones (and we can assume the same priority is applied to others).

Reply:
Added in [38532](https://github.com/elastic/elastic-otel-java/pull/370/commits/38532ee1b4dd0b59bee631e8e0aa75772d83e8f7)